Library/Homebrew/cmd/update-report.rb

```diff
@@ -43,6 +45,18 @@ def update_report
           safe_system "git", "config", "--local", "--replace-all", "homebrew.analyticsmessage", "true"
         end
       end
+
+      donation_message_displayed =
+        Utils.popen_read("git", "config", "--local", "--get", "homebrew.donationmessage").chuzzle
+      if donation_message_displayed != "true"
+        ohai "Homebrew is run entirely by unpaid volunteers. Please consider donating:"
+        puts "  #{Formatter.url("https://github.com/Homebrew/brew#donations")}\n"
+
+        # Consider the message possibly missed if not a TTY.
+        if $stdout.tty?
+          safe_system "git", "config", "--local", "--replace-all", "homebrew.donationmessage", "true"
```

Comment:
If never run on a TTY, for example in a non-interactive script, will this message be printed at every execution of `brew`?

Reply:
Every execution of `brew update`: yes. That's the same with the analytics messaging.